src/main/java/org/apache/sysml/hops/estim/EstimatorMatrixHistogram.java

```diff
@@ -31,254 +31,256 @@
+				nnz += (h1.cNnz[j] - h1.cNnz1e[j]) * h2.rNnz1e[j];
+				// approximate fraction, w/o double counting
+				double lsp = (double) (h1.cNnz[j] - h1.cNnz1e[j]) * (h2.rNnz[j] - h2.rNnz1e[j]) / mnOut;
+				spOutRest = spOutRest + lsp - spOutRest * lsp;
 			}
-			nnz += (long)(spOutRest * mnOut);
+			nnz += (long) (spOutRest * mnOut);
 		}
-		//general case with approximate output
+		// general case with approximate output
 		else {
-			long mnOut = h1.getRows()*h2.getCols();
+			long mnOut = h1.getRows() * h2.getCols();
 			double spOut = 0;
-			for( int j=0; j<h1.getCols(); j++ ) {
+			for (int j = 0; j < h1.getCols(); j++) {
 				double lsp = (double) h1.cNnz[j] * h2.rNnz[j] / mnOut;
-				spOut = spOut + lsp - spOut*lsp;
+				spOut = spOut + lsp - spOut * lsp;
 			}
-			nnz = (long)(spOut * mnOut);
+			nnz = (long) (spOut * mnOut);
 		}
-		
-		//exploit upper bound on nnz based on non-empty rows/cols
-		nnz = (h1.rNonEmpty >= 0 && h2.cNonEmpty >= 0) ?
-			Math.min((long)h1.rNonEmpty * h2.cNonEmpty, nnz) : nnz;
-		
-		//exploit lower bound on nnz based on half-full rows/cols
-		nnz = (h1.rNdiv2 >= 0 && h2.cNdiv2 >= 0) ?
-			Math.max((long)h1.rNdiv2 * h2.cNdiv2, nnz) : nnz;
-		
-		//compute final sparsity
-		return OptimizerUtils.getSparsity(
-			h1.getRows(), h2.getCols(), nnz);
+
+		// exploit upper bound on nnz based on non-empty rows/cols
+		nnz = (h1.rNonEmpty >= 0 && h2.cNonEmpty >= 0) ? Math.min((long) h1.rNonEmpty * h2.cNonEmpty, nnz) : nnz;
+
+		// exploit lower bound on nnz based on half-full rows/cols
+		nnz = (h1.rNdiv2 >= 0 && h2.cNdiv2 >= 0) ? Math.max((long) h1.rNdiv2 * h2.cNdiv2, nnz) : nnz;
+
+		// compute final sparsity
+		return OptimizerUtils.getSparsity(h1.getRows(), h2.getCols(), nnz);
 	}
-	
+
 	private static class MatrixHistogram {
 		// count vectors (the histogram)
-		private final int[] rNnz;    //nnz per row
-		private int[] rNnz1e = null; //nnz per row for cols w/ <= 1 non-zeros
-		private final int[] cNnz;    //nnz per col
-		private int[] cNnz1e = null; //nnz per col for rows w/ <= 1 non-zeros
+		private final int[] rNnz; // nnz per row
+		private int[] rNnz1e = null; // nnz per row for cols w/ <= 1 non-zeros
+		private final int[] cNnz; // nnz per col
+		private int[] cNnz1e = null; // nnz per col for rows w/ <= 1 non-zeros
 		// additional summary statistics
-		private final int rMaxNnz, cMaxNnz;     //max nnz per row/row
-		private final int rN1, cN1;             //number of rows/cols with nnz=1
-		private final int rNonEmpty, cNonEmpty; //number of non-empty rows/cols (w/ empty is nnz=0)
-		private final int rNdiv2, cNdiv2;       //number of rows/cols with nnz > #cols/2 and #rows/2
-		private boolean fullDiag;               //true if there exists a full diagonal of nonzeros
-		
+		private final int rMaxNnz, cMaxNnz; // max nnz per row/row
+		private final int rN1, cN1; // number of rows/cols with nnz=1
+		private final int rNonEmpty, cNonEmpty; // number of non-empty rows/cols
+												// (w/ empty is nnz=0)
```

Comment:
Please refrain from formatting existing code, especially via auto formatting as it (1) creates unnecessary merge conflicts, and (2) really does not look good.